crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1152,6 +1158,56 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
             )));
         }
 
+        // Check whether this version covers all supported platforms.
+        if !dist.implied_markers().is_true() {
+            for marker in [
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "linux".to_string(),
+                }),
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "win32".to_string(),
+                }),
+                MarkerTree::expression(MarkerExpression::String {
+                    key: MarkerValueString::SysPlatform,
+                    operator: MarkerOperator::Equal,
+                    value: "darwin".to_string(),
+                }),
+            ] {
+                // If the platform is part of the current environment...
+                if env.included_by_marker(marker) {
+                    // But isn't supported by the distribution...
+                    if dist.implied_markers().is_disjoint(marker) {
+                        // Then we need to fork.
+                        let forks = fork_version_by_marker(env, marker);
+                        return if forks.is_empty() {
+                            Ok(Some(ResolverVersion::Unavailable(
+                                candidate.version().clone(),
+                                UnavailableVersion::IncompatibleDist(IncompatibleDist::Wheel(
+                                    IncompatibleWheel::NoBinary,
```

Comment:
Oh sorry, this is a TODO -- I'll fix it up tomorrow (going to bed soon). I need to do proper error messages for this.